app/services/topology_service_api.rb

```diff
@@ -11,6 +11,10 @@ def initialize(options)
     @api_instance = TopologicalInventoryApiClient::DefaultApi.new
   end
 
+  def service_offering_ref
```

Comment:
@syncrou 
Who would call this function? Its predicated on the fact that there is a params['portfolio_item_id'] filled in?
its definition in the ToplogyServiceAPI also suggests that it can be inherited by sub classes, which might not have the portfolio_item_id defined. It seems like it belongs in the subclass.

Reply:
@mkanoor - I thought that param would always exist, this has been reverted.

Reply:
@syncrou 
The param will always be there, but will it have the portfolio_item_field?
When we interact with the topology service to get the status of a provisioning we have the order_item_id but no portfolio_item_id.

So my main concern with moving this to the parent class is that the parent class is making an assumption that param will always have a field called portfolio_item_id, which is not correct. It makes sense to have that function only in the subclass where it's relevant and its guaranteed that there is a portfolio_item_id.